raptor/raptor.py

```diff
@@ -25,42 +25,36 @@
 class Raptor(object):
     """Container class for Raptor"""
 
-    def __init__(self, options):
+    def __init__(self, app, binary):
         self.raptor_venv = os.path.join(os.getcwd(), 'raptor-venv')
         self.log = get_default_logger(component='raptor')
 
         self.control_server = None
-        self.browser = options.browser
-        self.browser_path = options.browser_path
-        self.test = options.test
+        self.app = app
+        self.binary = binary
         self.sysdir = os.path.abspath(os.getcwd())
 
-        pref_file = os.path.join(here, 'preferences', '{}.json'.format(self.browser))
+        pref_file = os.path.join(here, 'preferences', '{}.json'.format(self.app))
         prefs = {}
         if os.path.isfile(pref_file):
             with open(pref_file, 'r') as fh:
                 prefs = json.load(fh)
 
         try:
-            self.profile = create_profile(self.browser, preferences=prefs)
+            self.profile = create_profile(self.app, preferences=prefs)
         except NotImplementedError:
             self.profile = None
 
-    def verify_options(self):
-        self.log.info("TODO: Ensure cmd line options are valid before continuing i.e. test exists")
-
-    def gen_test_url(self):
-        gen_test_url(self.browser, self.test, self.sysdir)
-
     def install_webext(self):
-        install_webext(self.browser, self.profile)
+        install_webext(self.app, self.profile)
 
     def start_control_server(self):
         self.control_server = RaptorControlServer()
         self.control_server.start()
 
-    def run_test(self):
-        start_browser(self.browser, self.browser_path, self.profile)
+    def run_test(self, test):
+        gen_test_url(self.app, test, self.sysdir)
```

Comment:
I believe this won't work - gen_test_url has to be done before the WebExt is installed - as gen_test_url writes a background script into the web extension folder, that is then loaded when the web ext is installed.

There may have already been a /webext/raptor-firefox/augo_gen_settings_url.js file left over from before and that's why this worked.

The reason I write this file into the web extension, is that's the only way to get the test name into the extension (then it queries the control server for the info for that test name).

The other alternative is I pass the test name into the control server init - however I didn't do it that way as I thought it best to keep the control server instance independent of the test being run. 

However we are also creating the profile now at the time of Raptor init, so maybe it wouldn't hurt to tie in the test name to the control server init also in that we probably will just be using the same Raptor and control server instance once anyway.

Reply:
Ah, yeah sorry I didn't test this properly. Could we just also call `install_webext` implicitly in `run_test`? I'll re-push shortly with an example of what I mean.

As an aside, I think being able to run multiple tests from the same Raptor instance would prove useful in the future. Imagine this framework running tens of 'micro benchmarks' all in the same job. Having to re-create everything each time would be a waste. I think if Raptor is a success, it means adding new perf tests will be easy for developers to do, and in that world one test per job doesn't scale.

Reply:
Yes that works fine and actually I like that better, no need to split out each step into it's own method.

Good insight thanks and I agree. I wasn't sure if we would require a new totally clean browser profile before each test though (and we're currently creating the profile on Raptor init - however wouldn't be hard to just call create_profile again anyway).